lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -43,7 +43,7 @@ export async function generateLockFile(
   try {
     const npmToolConstraint: ToolConstraint = {
       toolName: 'npm',
-      constraint: config.constraints?.npm,
+      constraint: config.constraints?.npm ?? config.extractedConstraints?.npm,
```

Comment:
🤔 instead of using extracted constraints we should extract again in a follow up PR, so we also get potential updated version. we do that in other managers too

Reply:
Yes, but this is the simplest way to fix the bug right now